src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -584,7 +581,8 @@ static ucs_status_t uct_ib_mlx5_devx_dereg_atomic_key(uct_ib_md_t *ibmd,
     uct_ib_mlx5_mem_t *memh = ucs_derived_of(ib_memh, uct_ib_mlx5_mem_t);
     uct_ib_mlx5_md_t *md    = ucs_derived_of(ibmd, uct_ib_mlx5_md_t);
 
-    if (!(md->flags & UCT_IB_MLX5_MD_FLAG_KSM)) {
+    if (!ucs_test_all_flags(md->flags, UCT_IB_MLX5_MD_FLAG_KSM |
+                                       UCT_IB_MLX5_MD_FLAG_INDIRECT_ATOMICS)) {
```

Comment:
maybe we shall set UCT_IB_MLX5_MD_FLAG_INDIRECT_ATOMICS only if **both** KSM is enabled and umr_modify_atomic_disabled is 0?
to not have to check both flags every time

Reply:
It makes sense according to current usage but this is capability of MD which probably can be used in other way in future. Maybe define mask `UCT_IB_MLX5_MD_FLAG_KSM_ATOMICS = UCT_IB_MLX5_MD_FLAG_KSM | UCT_IB_MLX5_MD_FLAG_INDIRECT_ATOMICS`?

Reply:
i'm afraid mask may be mistaken and not used with test_all_flags
better define and inline helper..